### PR TITLE
Update transformation service

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1304

### Change description ###
Update transformation service to support both old and new transformation request format
- Get the exception record id from the respective field 
   - `id` for the old request
   -  `exception_record_id` for the new request

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/services/DocumentMapper.java [hunk 1]
```diff
@@ -14,9 +14,11 @@
      */
     public Item<ScannedDocument> toCaseDoc(
         InputScannedDoc exceptionRecordDoc,
-        String exceptionRecordReference
+        String exceptionRecordReference,
+        boolean isAutomatedProcess
     ) {
-        if (exceptionRecordDoc == null) {
+        if (!isAutomatedProcess && exceptionRecordDoc == null) {
```
Comment (aliveni) [inline src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/services/DocumentMapper.java:20]: Currently, when the exception record id is null, transformation result returns NO documents. 
But the Automated case creation request doesn't contain exception record id so transformation service should return scanned documents with no exception record id (which is done in the else part)

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/services/DocumentMapper.java [hunk 2]
```diff
@@ -14,9 +14,11 @@
...
+            // When request is NOT auto case creation and exception record reference is null
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/services/DocumentMapper.java:21]: should it not throw exception for this scenario ?

Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/services/DocumentMapper.java:21]: Yeah, it should. But it was implemented like this before. I'll update `ExceptionRecordValidator.java` to assert that exception record id exists. is that ok?

Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/services/DocumentMapper.java:21]: I misread the parameter name `exceptionRecordDoc` as `exceptionRecordId`. This change is not required here, I'll revert

**Pull request merged**